Firestore/core/test/firebase/firestore/remote/serializer_test.cc

```diff
@@ -1488,7 +1483,49 @@ TEST_F(SerializerTest, EncodesListenRequestLabels) {
   }
 }
 
-TEST_F(SerializerTest, DecodesListenResponseWithTargetChange) {
+TEST_F(SerializerTest, DecodesMutationResult) {
+  std::vector<FieldValue> transformations({FieldValue::FromBoolean(true),
+                                           FieldValue::FromInteger(1234),
+                                           FieldValue::FromString("string")});
+  MutationResult model(Version(123456789), std::move(transformations));
+
+  v1::WriteResult proto;
+
+  proto.mutable_update_time()->set_seconds(
+      Version(123456789).timestamp().seconds());
+  proto.mutable_update_time()->set_nanos(
+      Version(123456789).timestamp().nanoseconds());
+  proto.mutable_transform_results()->Add();
+  proto.mutable_transform_results()->Add();
+  proto.mutable_transform_results()->Add();
+  (*proto.mutable_transform_results())[0] = ValueProto(true);
+  (*proto.mutable_transform_results())[1] = ValueProto(1234);
+  (*proto.mutable_transform_results())[2] = ValueProto("string");
+
+  SCOPED_TRACE("DecodesMutationResult");
+  ExpectDeserializationRoundTrip(model, proto, Version(10000000));
+}
+
+TEST_F(SerializerTest, DecodesMutationResultWithNoUpdateTime) {
+  std::vector<FieldValue> transformations({FieldValue::FromBoolean(true),
+                                           FieldValue::FromInteger(1234),
+                                           FieldValue::FromString("string")});
+  MutationResult model(Version(10000000), std::move(transformations));
+
+  v1::WriteResult proto;
+
+  proto.mutable_transform_results()->Add();
```

Comment:
Objective-C doesn't add any transformations in this case, which implicitly tests for the case when the array is empty. Perhaps do the same, or reduce the number of transformations to 1?

Reply:
Done.